Tasks/ServiceFabricDeploy/deploy.ps1

```diff
@@ -0,0 +1,87 @@
+    Write-Host (Get-VstsLocString -Key SearchingForPublishProfile -ArgumentList $publishProfilePathSearchPattern) 
+    $publishProfilePath = Find-VstsFiles -LegacyPattern $publishProfilePathSearchPattern
+    Assert-SingleItem $publishProfilePath $publishProfilePathSearchPattern
+    Assert-VstsPath -LiteralPath $publishProfilePath -PathType Leaf
+    Write-Host (Get-VstsLocString -Key FoundPublishProfile -ArgumentList $publishProfilePath)
+    
+    $applicationPackagePathSearchPattern = Get-VstsInput -Name applicationPackagePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForApplicationPackage -ArgumentList $applicationPackagePathSearchPattern)
+
+    # Workaround for bug in Find-VstsFiles that doesn't handle the IncludeDirectories parameter correctly. Modifies the search pattern to look for app manifest file instead of the directory
+    # and then get the directory of that file.
+    $applicationPackagePathSearchPattern = [System.IO.Path]::Combine($applicationPackagePathSearchPattern, "ApplicationManifest.xml")
+    $applicationPackagePath = Find-VstsFiles -LegacyPattern $applicationPackagePathSearchPattern
+    Assert-SingleItem $applicationPackagePath $applicationPackagePathSearchPattern
+    $applicationPackagePath = [System.IO.Path]::GetDirectoryName($applicationPackagePath)
+    Assert-VstsPath -LiteralPath $applicationPackagePath -PathType Container
+    Write-Host (Get-VstsLocString -Key FoundApplicationPackage -ArgumentList $applicationPackagePath)
+
+    $aadUserName = Get-VstsInput -Name aadUserName -Require
+    Write-Host "userName: $aadUserName"
+    
+    $aadUserPassword = Get-VstsInput -Name aadUserPassword -Require
+    # Do NOT write value of password to the output
+
+    $publishProfile = Read-PublishProfile $publishProfilePath
+    $clusterConnectionParameters = $publishProfile.ClusterConnectionParameters
+    
+    if (-not $clusterConnectionParameters.ContainsKey("AzureActiveDirectory") -or $clusterConnectionParameters.Item("AzureActiveDirectory") -eq $false)
+    {
+        throw (Get-VstsLocString -Key AzureActiveDirectoryNotSet)
+    }
+    
+    # Connect to cluster
+    $securityToken = Get-AadSecurityToken -ClusterConnectionParameters $clusterConnectionParameters -UserName $aadUserName -UserPassword $aadUserPassword
+    $clusterConnectionParameters["SecurityToken"] = $securityToken
+    $clusterConnectionParameters["WarningAction"] = "SilentlyContinue"
+    [void](Connect-ServiceFabricCluster @clusterConnectionParameters)
+    Write-Host (Get-VstsLocString -Key ConnectedToCluster)
+    
+    $regKey = "HKLM:\SOFTWARE\Microsoft\Service Fabric SDK"
+    $moduleFolderPath = (Get-ItemProperty -Path $regKey -Name FabricSDKPSModulePath).FabricSDKPSModulePath
+
+    if (Test-Path "$moduleFolderPath\ServiceFabricSDK.ps1")
+    {
+        . "$moduleFolderPath\ServiceFabricSDK.ps1"
+    }
+    else
+    {
+        foreach ($scriptFile in Get-ChildItem $moduleFolderPath -Filter "*.ps1")
```

Comment:
There is but it can't be used in this context.  It's due to the way in which PowerShell scopes its variables.  The functions contained in the modules need access to a global connection object variable which only works if you dot source the file versus importing.